pkg/controllers/provisioning/v1alpha1/reallocator/filter.go

```diff
@@ -25,40 +28,75 @@ type Filter struct {
 	kubeClient client.Client
 }
 
-// Gets Nodes fitting some underutilization
-// TODO: add predicate
-func (f *Filter) GetUnderutilizedNodes(ctx context.Context) ([]*v1.Node, error) {
-	nodeList := &v1.NodeList{}
+// Gets Nodes fitting some underutilization predicate
+func (f *Filter) GetUnderutilizedNodes(ctx context.Context, provisioner *v1alpha1.Provisioner) ([]*v1.Node, error) {
 	underutilized := []*v1.Node{}
 
-	// 1. Get all nodes
-	if err := f.kubeClient.List(ctx, nodeList); err != nil {
-		return nil, fmt.Errorf("listing nodes, %w", err)
+	// 1. Get all provisioner labeled nodes
+	nodeList, err := f.getNodes(ctx, provisioner)
+	if err != nil {
+		return nil, err
 	}
 
 	// 2. Get nodes and the pods on each node
 	for _, node := range nodeList.Items {
 		pods, err := f.getPodsOnNode(ctx, node.Name)
 		if err != nil {
-			return []*v1.Node{}, fmt.Errorf("filtering pods on nodes, %w", err)
+			return []*v1.Node{}, err
 		}
-		if f.isUnderutilized(&node, pods) {
+		// Only checks if it has 0 non-daemon pods right now
```

Comment:
does it need to check more in future? (should there be a TODO here?)

Reply:
Yeah. There's a TODO at the higher level right now.